src/python/pants/backend/python/typecheck/pytype/subsystem.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.target_types import ConsoleScript
+from pants.core.util_rules.config_files import ConfigFilesRequest
+from pants.option.option_types import ArgsListOption, FileOption, SkipOption
+from pants.util.strutil import help_text, softwrap
+
+
+class Pytype(PythonToolBase):
+    options_scope = "pytype"
+    name = "Pytype"
+    help = help_text(
+        """
+        The Pytype utility for typechecking Python code
+        (https://github.com/google/pytype).
+        """
+    )
+
+    register_interpreter_constraints = True
+    default_interpreter_constraints = ["CPython>=3.7,<3.10"]
+
+    default_main = ConsoleScript("pytype")
+    default_requirements = ["pytype==2023.6.16"]
```

Comment:
Question: To generate the lockfile, I had to locally change [this line](https://github.com/pantsbuild/pants/blob/main/build-support/bin/generate_builtin_lockfiles.py#L63) to `>=3.7,<3.11`. Is there a way to specify this restriction on the subsystem.py file instead?

Reply:
You can provide it per tool, so in the `all_python_tools` tuple, you'd have:
```python
    PythonTool(Pytype, "pants.backend.python.typecheck.pytype", interpreter_constraints=">=3.7,<3.11"),
```

to override the default set here: https://github.com/pantsbuild/pants/blob/9f49668dcfb3cc9ae877850d14e66dbe7045f3df/build-support/bin/generate_builtin_lockfiles.py#L88-L89

Reply:
Thanks, updated.